#!/usr/bin/env python
PACKAGE = "ras"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_vision_range", double_t, 0, "maximum distance to obstacles", 100.0, 0.0, 200.0)
gen.add("min_vision_range", double_t, 0, "minimum distance to obstacles", 3.0, 0.0, 50.0)
gen.add("ego_name", str_t, 0, "ego_vehicle name in carla world", "ego_vehicle")
gen.add("keep_time", double_t, 0, "acceleration coefficient", 2.0, 0.0, 10.0)
gen.add('conservative', bool_t, 0, 'start conservative mode', True)
gen.add('detection_level_thres', double_t, 0, 'threshold to cut detection level', 20.0, 0.0, 100)

intervene_enum = gen.enum([gen.const('control', int_t, 0, 'Controller'),
                            gen.const('enter', int_t, 1, 'Enter key'),
                            gen.const('touch', int_t, 2, 'Touch Display')],
                            'enum to set intervene type')
gen.add('intervene_type', int_t, 0, 'intervene_type selection', 0, 0, 2, edit_method=intervene_enum)

exit(gen.generate(PACKAGE, "ras", "ras"))
